client/src/components/Results/ResultsMap.js

```diff
@@ -127,6 +175,77 @@ function Map({
     );
   }
 
+  const data = {
+    type: "FeatureCollection",
+    features: (stakeholders || [])
+      .filter(
+        (sh) =>
+          sh.latitude && sh.longitude && !(sh.inactive || sh.inactiveTemporary)
+      )
+      .map((stakeholder) => {
+        return {
+          type: "Feature",
+          properties: {
+            stakeholderId: stakeholder.id,
+          },
+          geometry: {
+            type: "Point",
+            coordinates: [stakeholder.longitude, stakeholder.latitude, 0],
+          },
+        };
+      }),
+  };
+
+  const onMapClick = (e) => {
+    // if we are clicking on a feature
+    if (e.features && e.features.length > 0) {
+      const layer = e.features[0].layer.id;
+
+      // if we are clicking on a cluster, then zoom in on it
+      if (layer === "clusters" || layer === "cluster-count") {
+        const feature = e.features[0];
+        const clusterId = feature.properties.cluster_id;
+
+        const mapboxSource = sourceRef.current.getSource();
+
+        mapboxSource.getClusterExpansionZoom(clusterId, (err, zoom) => {
+          if (err) {
+            return;
+          }
+
+          setViewport({
+            ...viewport,
+            longitude: feature.geometry.coordinates[0],
+            latitude: feature.geometry.coordinates[1],
+            zoom,
+            transitionDuration: 500,
+          });
+        });
+
+        return;
+      }
+    }
+
+    doSelectStakeholder(null);
+  };
+
+  // the list of stake holders that are not in a cluster
+  const shownStakeHolders = [];
```

Comment:
should be `shownStakeholders` "stakeholder" is a single word so no need to camelCase it.